src/main/java/com/github/kaktushose/jda/commands/dispatching/events/Event.java

```diff
@@ -0,0 +1,208 @@
+    private final Runtime runtime;
+
+    /// Constructs a new Event.
+    ///
+    /// @param event               the subtype [T] of [GenericInteractionCreateEvent]
+    /// @param interactionRegistry the corresponding [InteractionRegistry]
+    /// @param runtime             the [Runtime] this event lives in
+    protected Event(@NotNull T event, @NotNull InteractionRegistry interactionRegistry, @NotNull Runtime runtime) {
+        this.event = event;
+        this.interactionRegistry = interactionRegistry;
+        this.runtime = runtime;
+    }
+
+    /// Returns the underlying [GenericInteractionCreateEvent] of this event
+    ///
+    /// @return the [GenericInteractionCreateEvent]
+    @NotNull
+    public T jdaEvent() {
+        return this.event;
+    }
+
+    /// Gets a [`Button`][com.github.kaktushose.jda.commands.annotations.interactions.Button] based on the method name
+    /// and transforms it into a JDA [Button].
+    ///
+    /// The button will be linked to the current [Runtime]. This may be useful if you want to send a component without
+    /// using the framework.
+    ///
+    /// @param button the name of the button
+    /// @return the JDA [Button]
+    @NotNull
+    public Button getButton(@NotNull String button) {
+        if (!button.matches("[a-zA-Z]+\\.[a-zA-Z]+")) {
+            throw new IllegalArgumentException("Unknown Button");
+        }
+
+        String sanitizedId = button.replaceAll("\\.", "");
+        ButtonDefinition buttonDefinition = interactionRegistry.getButtons().stream()
+                .filter(it -> it.getDefinitionId().equals(sanitizedId))
+                .findFirst().orElseThrow(() -> new IllegalArgumentException("Unknown Button"));
+
+        return buttonDefinition.toButton().withId(buttonDefinition.boundCustomId(runtimeId()));
+    }
+
+    /// Gets a [StringSelectMenu] or [EntitySelectMenu] based on the method name and transforms it into a JDA [SelectMenu].
+    ///
+    /// The select menu will be linked to the current [Runtime]. This may be useful if you want to send a component
+    /// without using the framework.
+    ///
+    /// @param <S>  the type of [SelectMenu]
+    /// @param menu the name of the select menu
+    /// @return the JDA [SelectMenu]
+    @SuppressWarnings("unchecked")
+    @NotNull
+    public <S extends SelectMenu> S getSelectMenu(@NotNull String menu) {
+        if (!menu.matches("[a-zA-Z]+\\.[a-zA-Z]+")) {
+            throw new IllegalArgumentException("Unknown Select Menu");
+        }
+
+        String sanitizedId = menu.replaceAll("\\.", "");
+        GenericSelectMenuDefinition<?> selectMenuDefinition = interactionRegistry.getSelectMenus().stream()
+                .filter(it -> it.getDefinitionId().equals(sanitizedId))
+                .findFirst().orElseThrow(() -> new IllegalArgumentException("Unknown Select Menu"));
+
+        return (S) selectMenuDefinition.toSelectMenu(runtimeId(), true);
+    }
```

Comment:
I see no point in making this generic.
```suggestion
    public SelectMenu getSelectMenu(@NotNull String menu) {
        if (!menu.matches("[a-zA-Z]+\\.[a-zA-Z]+")) {
            throw new IllegalArgumentException("Unknown Select Menu");
        }

        String sanitizedId = menu.replaceAll("\\.", "");
        GenericSelectMenuDefinition<?> selectMenuDefinition = interactionRegistry.getSelectMenus().stream()
                .filter(it -> it.getDefinitionId().equals(sanitizedId))
                .findFirst().orElseThrow(() -> new IllegalArgumentException("Unknown Select Menu"));

        return selectMenuDefinition.toSelectMenu(runtimeId(), true);
    }
```

Reply:
It's needed because JDAs SelectMenu has two subtypes you might explicitly want

Reply:
The user has to cast it anyway, no matter if it's generic or not